### PR TITLE
feat: custom objects, entities, associations

### PR DESCRIPTION
**Pull request by albertyfwu**
Change list (all of these are supported for Salesforce and HubSpot):
- Stop supporting custom objects for SyncConfig (no more syncing custom objects - need to use Entities instead)
- Allow mapping Entities to custom objects (FE and BE)
- Move AssociationTypes and Custom Objects to new Metadata API
- Support listing custom and standard objects from providers in Metadata API
- Support creating AssociationTypes between Entities
- Support creating Associations between Entity Records
- Update docs to show Metadata API
- Make creation of AssociationTypes and Entities more unified so that we perform provider-specific mutations; for example, if you pass in a name like `related_opportunity` to create an AssociationType in Salesforce, we will append `__c` to it to make sure it doesn't get rejected by the API
- moved properties endpoint to Metadata API
- changed `name` to `suggestedName` or something for creation of AssociationTypes and Custom Objects to make it clear that Supaglue may append `__c` to the name to make it work for Salesforce.
- ...other changes
- 
image

## Test Plan

[Describe test plan here]

## Deployment instructions

[Add any special deployment instructions here]

### CODE REVIEW

**Review by asdfryan — APPROVED** [review]

File: openapi/v2/actions/components/schemas/objects/entity_or_object_record.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+oneOf:
```
Comment (tomkit) [inline openapi/v2/actions/components/schemas/objects/entity_or_object_record.yaml:1]: Can you add one full example here

File: openapi/v2/actions/components/schemas/objects/entity_or_object.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+oneOf:
```
Comment (tomkit) [inline openapi/v2/actions/components/schemas/objects/entity_or_object.yaml:1]: Can you add one full example here

File: openapi/v2/actions/components/schemas/objects/custom_object_record.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+type: object
+properties:
+  id:
+    type: string
+  custom_object_id:
+    type: string
+  data:
+    type: object
+    properties:
```
Comment (tomkit) [inline openapi/v2/actions/components/schemas/objects/custom_object_record.yaml:9]: can you add an example here

File: openapi/v2/actions/paths/objects/custom@{object_id}@{record_id}.yaml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  security:
+    - x-api-key: []
+  responses:
+    200:
+      description: Custom object record
+      content:
+        application/json:
+          schema:
+            $ref: ../../components/schemas/objects/custom_object_record.yaml
+          examples:
```
Comment (tomkit) [inline openapi/v2/actions/paths/objects/custom@{object_id}@{record_id}.yaml:15]: Can we omit `examples: Example: value:` going forward and just use schema examples

File: openapi/v2/metadata/components/schemas/create_update_custom_object.yaml [hunk 5]
```diff
@@ -1,9 +1,9 @@
 type: object
 properties:
-  name:
+  suggested_name:
     type: string
     example: ticket
-    description: In Salesforce, this must end with `__c`.
+    description: The name you'd like to use for the custom object. For some providers, like Salesforce, we might mutate the name, like append `__c`.
```
Comment (tomkit) [inline openapi/v2/metadata/components/schemas/create_update_custom_object.yaml:6]: Can we enumerate the known ones today (Salesforce, Hubspot)

File: openapi/v2/actions/components/schemas/objects/entity_or_object_record.yaml (hunk 1 above)
Comment (albertyfwu) [reply openapi/v2/actions/components/schemas/objects/entity_or_object_record.yaml:1]: looks like we don't use this anywhere. going to delete this yaml

File: openapi/v2/actions/components/schemas/objects/entity_or_object.yaml (hunk 2 above)
Comment (albertyfwu) [reply openapi/v2/actions/components/schemas/objects/entity_or_object.yaml:1]: this file not used either. deleting.

**Pull request merged**